### ISSUE TITLE
Hyperparameter settings for FB15K_237

### ISSUE THREAD

**Issue by Diqingling** [Sujit-O/pykg2vec#173]
Are there any other hyperparameter settings for different datasets to reproduce their performance in the original paper,such as FB15K_237? I tried to use the hyperparameters for FB15K to train FB15K_237 but the results are far below the benchmarks. The process of adjusting parameters is very time-consuming so I want to ask if anyone can provide some settings.
Thanks.

**Comment by baxtree** [comment]
Hi, @Diqingling. Looks like the original [FB15K_237 paper](https://www.aclweb.org/anthology/W15-4007/) has no code published and it does not mention things like learning rate for `model E` and `DistMult` they used either. In this case, we encourage users to explore hyperparameter tuning and you can find how-to in [here](https://pykg2vec.readthedocs.io/en/latest/auto_examples/index.html). Meanwhile, we are extracting the golden hyperparameters for [ConvKB](https://github.com/daiquocnguyen/ConvKB) which also used FB15K_237 and of course, nothing stops you from extracting them yourself and create your own custom yaml file.

**Comment by Diqingling** [comment]
Thank you. I’ll try to tune the hyperparameters. Another question is that since the hyperparameters in the yaml files don’t fit dataset FB15K_237, I used them to train models with FB15K as well. However, the filtered hit-10 result of TransE is around 60.2% ,the benchmarks given in [RotatE](https://arxiv.org/pdf/1902.10197.pdf)   is 74.9%. So I want to ask if there is something wrong with my training process or with the hyperparameters settings?

**Comment by Diqingling** [comment]
Thank you. I’ll try to tune the hyperparameters. Another question is that since the hyperparameters in the yaml files don’t fit dataset FB15K_237, I used them to train models with FB15K as well. However, the filtered hit-10 result of TransE is around 60.2% ,the benchmark given in  [RotatE](https://arxiv.org/pdf/1902.10197.pdf) is 74.9%. So I want to ask if there is something wrong with my training process or with the hyperparameters settings?

**Comment by baxtree** [comment]
Probably there was nothing wrong with your training process coz the best result we can get so far is 60.7%:
Model | MR/FMR | MRR/FMRR | HIT-1 | HIT-3 | HIT-5 | HIT-10
--- | --- | --- | --- |--- |--- |---
TransE | 215.1193/77.7292 | 0.1921/0.3413 | 0.0852/0.2056 | 0.2176/0.4050 | 0.3003/0.4921 | 0.4206/**0.6070**

TransE is one of the simplest models and if there is an implementation bug, it shall be easy to spot. Can you compare hyperparameters set in [TransE.yaml](https://github.com/Sujit-O/pykg2vec/blob/master/pykg2vec/hyperparams/TransE.yaml) with those displayed in your "Global Setting" log during the training?

**Comment by Diqingling** [comment]
I checked the hyperparameters in the log file and they match with the settings in TransE.yaml. The difference between 60.2% and 60.7% may be normal volatility during training process. However they are both far below the results given in [HoLE](https://arxiv.org/pdf/1510.04935.pdf) paper ,where the filtered hit-10 is 74.9%. Does it mean the hyperparameters in TransE.yaml may not be golden hyperparameters? What’s more ,how do you get the hyperparameters in the exsisting yaml files? Using hyperparameter tuning module in the project or taking them from the original model papers directly?

**Comment by baxtree** [comment]
All hyperparameters are derived from the original [paperswithcode](https://paperswithcode.com/) and there might be a slim chance that some authors later corrected errors in their published work, which have not been incorporated in pykg2vec. For the previous [tensorflow](https://github.com/Sujit-O/pykg2vec/tree/tf2-master) implementation, you can see `HIT-10` is 66% from that table. Hi, @louisccc, would you like to weigh in and share some history about this?
